docs/design/RUC-on-type.md

```diff
@@ -0,0 +1,189 @@
+# RequiresUnreferencedCode on Types
```

Comment:
Let's make this a spec for `RequiresUnreferencedCode`, all together. No need to just talk about types.

Reply:
I would not block on this for now - we need to resolve RUC-on-type behavior soon, so that we can implement it for .NET 6. Documenting the existing behavior of RUC on methods is definitely something we should do, but not part of this particular effort.